rxfingerprint/src/main/java/com/mtramin/rxfingerprint/FingerprintEncryptionObservable.java

```diff
@@ -115,7 +128,7 @@ protected void onAuthenticationSucceeded(Subscriber<FingerprintEncryptionResult>
 			subscriber.onNext(new FingerprintEncryptionResult(FingerprintResult.AUTHENTICATED, null, encryptedString));
 			subscriber.onCompleted();
 		} catch (CryptoDataException | IllegalBlockSizeException | BadPaddingException | InvalidParameterSpecException | UnsupportedEncodingException e) {
-			subscriber.onError(e);
+			checkKey(e, subscriber, () -> subscriber.onNext(new FingerprintEncryptionResult(FingerprintResult.INVALIDATED, null, null)));
```

Comment:
I'm unsure if the `FingerprintResult.INVALIDATED` gives the user of the library any additional information.

Once the key is invalidated this will be notified by the `onError` callback containing the `KeyPermanentlyInvalidatedException`. Consumers should then clear their stored data and it we should be able to simply remove/recreate the key if we receive `KeyPermanentlyInvalidatedException` in the `encrypt` call.

For that case I don't see the added benefit from having an additional `FingerprintResult`. What do you think?